include/zephyr/senss/senss_datatypes.h

```diff
@@ -0,0 +1,313 @@
+ * 0x00			       1 * 10E0
+ * 0x01			       1 * 10E1
+ * 0x02			       1 * 10E2
+ * 0x03			       1 * 10E3
+ * 0x04			       1 * 10E4
+ * 0x05			       1 * 10E5
+ * 0x06			       1 * 10E6
+ * 0x07			       1 * 10E7
+ * 0x08			       1 * 10E-8
+ * 0x09			       1 * 10E-7
+ * 0x0A			       1 * 10E-6
+ * 0x0B			       1 * 10E-5
+ * 0x0C			       1 * 10E-4
+ * 0x0D			       1 * 10E-3
+ * 0x0E			       1 * 10E-2
+ * 0x0F			       1 * 10E-1
+ *
+ * So, we can have below data present ranges:
+ *
+ * int8 type v:    [-128, 127] *10^[-8, 7]
+ * uint8 type v:   [0,	255] * 10^[-8, 7]
+ * int16 type v:   [-32768, 32767] * 10^[-8, 7]
+ * uint16 type v:  [0,	65535\r] * 10^[-8, 7]
+ * int32 type v:   [-2147483648,  2147483647] * 10^[-8, 7]
+ * uint32 type v:  [0,	4294967295] * 10^[-8, 7]
+ *
+ * To simple the data structure definition and save store memory, only keep v in code definitions,
+ * scale exponent x will defined in doc and spec,  but not explicitly present in code, for scenarios
+ * which need transfer to decimal value, such as in a algorithm process, need base on the sensor
+ * type and according the doc/spec to get the right scale exponent value x.
+ *
+ * A example in doc and spec can be like:
+ * 3D accelerometer:
+ *	Data Fields    Type	Unit	    Unit Exponent   Typical Range     Description
+ *	data[0]        int32	micro g     -6		    +/-4*10^6	      x axis acceleration
+ *	data[1]        int32	micro g     -6		    +/-4*10^6	      y axis acceleration
+ *	data[2]        int32	micro g     -6		    +/-4*10^6	      z axis acceleration
+ *
+ * Ambient Light:
+ *	Data Fields    Type	Unit	    Unit Exponent   Typical Range     Description
+ *	data[0]        uint32	milli lux   -3		    [0, 10000] *10^3  Ambient light lux level
+ *
+ * The complete doc/spec should describe all supported sensors like above example.
+ */
+
+/**
+ * @brief Sensor value data structure for quaternion.
+ * int16 version.
+ */
+struct senss_sensor_value_quaternion_int16 {
+	struct senss_sensor_value_header header;
+	struct {
+		uint32_t timestamp_delta;
+		int16_t  x;
+		int16_t  y;
+		int16_t  z;
+		int16_t  w;
+	} readings[1];
+};
+
+/**
+ * @brief Sensor value data structure for 3-aixs sensors.
+ * int32 version
+ */
+struct senss_sensor_value_3d_int32 {
+	struct senss_sensor_value_header header;
+	struct {
+		uint32_t timestamp_delta;
+		union {
+			int32_t v[3];
+			struct {
+				int32_t x;
+				int32_t y;
+				int32_t z;
+			};
+		};
+	} readings[1];
+};
+
+/**
+ * @brief Sensor value data structure for un-calibrated 3-aixs sensors.
+ * int32 version
+ */
+struct senss_sensor_value_uncalib_3d_int32 {
+	struct senss_sensor_value_header header;
+	struct {
+		uint32_t timestamp_delta;
+		union {
+			int32_t v[3];
+			struct {
+				int32_t x;
+				int32_t y;
+				int32_t z;
+			};
+		};
+		union {
+			int32_t bias[3];
+			struct {
+				int32_t bias_x;
+				int32_t bias_y;
+				int32_t bias_z;
+			};
+		};
+	} readings[1];
+};
```

Comment:
Regarding to time stamp, see discussion in the `senss_datatypes.h`

For the few questions,  yes,  its for SENSOR_TYPE_UNCALIBRATED_X. 
unlike `CHRE` use a separated bias event callback, `senss` will use unclibrated sensor type, and send bias data with sensor data together, this can get sensor data and bias data aligned, and not need new event callback for bias data event.

Reply:
But bias is unlikely to change, why add it to every data?

Reply:
Some sensors,  such as the magnetic sensor, the bias data is computed by the algorithm at runtime on each sensor data, the worst case is the bias data is different on each sensor data, that's why we put bias data together with sensor data.

And using a un-calibrated sensor type, there're many benefits, such as we can use single and common sensor concepts and infrastructures, for example, sensor subsystem use totally same common data/config path for all sensors, no need special path for bias (separated callback etc), and the batch & buffer, can be easy used also on bias data.

Reply:
> Some sensors, such as the magnetic sensor, the bias data is computed by the algorithm at runtime on each sensor data, the worst case is the bias data is different on each sensor data, that's why we put bias data together with sensor data.
> 
> And using a un-calibrated sensor type, there're many benefits, such as we can use single and common sensor concepts and infrastructures, for example, sensor subsystem use totally same common data/config path for all sensors, no need special path for bias (separated callback etc), and the batch & buffer, can be easy used also on bias data.

Continued discussion in #48150